src/main/scala/is/hail/asm4s/GenericTypeInfo.scala

```diff
@@ -58,6 +60,8 @@ final case class GenericTypeInfo[T : TypeInfo]() extends MaybeGenericTypeInfo[T]
       Code.boxBoolean(x.asInstanceOf[Code[Boolean]])
     case _: CharInfo.type =>
       Code.newInstance[java.lang.Character, Char](x.asInstanceOf[Code[Char]])
+    case _: UnitInfo.type =>
+      Code(x, Code._null[java.lang.Void])
```

Comment:
Ah, shouldn't Unit/Void take up no stack slots, in which case this is just x.asInstanceOf[Code[Unit]]?

Also, I'm slightly confused how Unit parameters are showing up.

Reply:
Thanks for the explanation, this all makes sense now.  I'm on board.